chain/client/src/chunk_inclusion_tracker.rs

```diff
@@ -57,42 +99,47 @@ impl ChunkInclusionTracker {
     }
 
     /// Function to return the chunks that are ready to be included in a block.
-    /// We filter out the chunks that are produced by banned chunk producers.
-    /// Return type contains some extra information needed for debug logs and metrics.
-    ///     HashMap from [shard_id] to (chunk_header, chunk received_time, chunk_producer account_id)
+    /// We filter out the chunks that are produced by banned chunk producers or have insufficient
+    /// chunk validator endorsements.
+    /// Return HashMap from [shard_id] -> chunk_hash
+    /// We can later use the chunk_hash to fetch chunk_header, chunk_endorsements, chunk_producer_and_received_time
     pub fn get_chunk_headers_ready_for_inclusion(
-        &self,
+        &mut self,
         epoch_id: &EpochId,
         prev_block_hash: &CryptoHash,
-    ) -> HashMap<ShardId, (ShardChunkHeader, chrono::DateTime<chrono::Utc>, AccountId)> {
-        self.prev_block_to_chunk_headers_ready_for_inclusion
-            .peek(prev_block_hash)
-            .cloned()
-            .unwrap_or_default()
-            .into_iter()
-            .filter(|(_, (chunk_header, _, chunk_producer))| {
-                let banned = self
-                    .banned_chunk_producers
-                    .contains(&(epoch_id.clone(), chunk_producer.clone()));
-                if banned {
-                    tracing::warn!(
-                        target: "client",
-                        chunk_hash = ?chunk_header.chunk_hash(),
-                        ?chunk_producer,
-                        "Not including chunk from a banned validator");
-                    metrics::CHUNK_DROPPED_BECAUSE_OF_BANNED_CHUNK_PRODUCER.inc();
+        chunk_validator: &mut ChunkValidator,
+    ) -> Result<HashMap<ShardId, ChunkHash>, Error> {
+        let mut chunk_headers_ready_for_inclusion = HashMap::new();
+        if let Some(entry) = self.prev_block_to_chunk_hash_ready.get_mut(prev_block_hash) {
```

Comment:
nit to reduce the nesting: 
``` let Some(entry) = ... else { return ... }  ```

Reply:
Aah! I had been trying to do that! Didn't know the syntax. Thanks!!